lib/tennis.rb

```diff
@@ -1,3 +1,54 @@
-require 'game'
+require 'pry'
+class Game
+  def initialize(player_one, player_two)
+    @player_one = player_one
+    @player_two = player_two
 
-puts "hi"
+    @player_one_counter = 0
+    @player_two_counter = 0
+  end
+
+  attr_accessor :player_one, :player_two, :player_one_counter, :player_two_counter
+
+  SCORES = ['0', '15', '30', '40', 'Adv', 'Win']
+
+  def current_score
+    "#{player_one} #{SCORES[player_one_counter]} - #{SCORES[player_two_counter]} #{player_two}"
+  end
+
+  def win_condition(input)
+
+    if input == 1 && @player_one_counter <= 4
+      if @player_one_counter < 3
```

Comment:
You could simplify this:

```
if input == 1
  player_one_counter += 1 if player_one_counter < 3
  determine_if_player_has_won_from_forty(player_one_counter, player_two_counter)
  determine_if_player_has_won_from_advantage(player_one_counter, player_two_counter)
end
```

This would create two private methods which pass in the counters. This means you can reuse the method when player two wins the point.

You could also potentially add boolean methods to determine whether the score is `deuce?` or whether a player is at `advantage?`

Also another minor point, when referring to accessors you don't need the `@` set before them after they've been initialized.